pyat/at/latticetools/observables.py

```diff
@@ -0,0 +1,974 @@
+            target:         Target value for a constraint. If :py:obj:`None`
+              (default), the residual will always be zero.
+            weight:         Weight factor: the residual is
+              :pycode:`((value-target)/weight)**2`
+            bounds:         Tuple of lower and upper bounds. The parameter
+              is constrained in the interval
+              [*target*\ +\ *low_bound* *target*\ +\ *up_bound*]
+
+        The *target*, *weight* and *bounds* inputs must be broadcastable to the
+        shape of *value*.
+
+        Example:
+
+            >>> obs = OrbitObservable(at.Monitor, axis=0)
+
+            Observe the horizontal closed orbit at monitor locations
+        """
+        name = self._set_name(name, "orbit", axis_(axis, "code"))
+        fun = _ArrayAccess(axis_(axis, "index"))
+        needs = {Need.ORBIT}
+        super().__init__(fun, refpts, needs=needs, name=name, **kwargs)
+
+
+class MatrixObservable(_ElementObservable):
+    """Observe the closed orbit at selected locations.
+
+    Processs the result of calling :py:func:`.find_m44` or :py:func:`.find_m44`
+    depending upon :py:meth:`~.Lattice.is_6d`.
+    """
+
+    def __init__(
+        self,
+        refpts: Refpts,
+        axis: AxisDef = Ellipsis,
+        name: Optional[str] = None,
+        **kwargs,
+    ):
+        # noinspection PyUnresolvedReferences
+        r"""
+        Args:
+            refpts:         Observation points.
+              See ":ref:`Selecting elements in a lattice <refpts>`"
+            axis:           Index in the transfer matrix, If :py:obj:`Ellipsis`,
+              the whole matrix is specified
+            name:           Observable name. If :py:obj:`None`, an explicit
+              name will be generated
+
+        Keyword Args:
+            statfun:        Post-processing function called on the value of the
+              observable. Example: :pycode:`statfun=numpy.mean`
+            target:         Target value for a constraint. If :py:obj:`None`
+              (default), the residual will always be zero.
+            weight:         Weight factor: the residual is
+              :pycode:`((value-target)/weight)**2`
+            bounds:         Tuple of lower and upper bounds. The parameter
+              is constrained in the interval
+              [*target*\ +\ *low_bound* *target*\ +\ *up_bound*]
+
+        The *target*, *weight* and *bounds* inputs must be broadcastable to the
+        shape of *value*.
+
+        Example:
+
+            >>> obs = MatrixObservable(at.Monitor, axis=('x', 'px'))
+
+            Observe the transfer matrix from origin to monitor locations and
+            extract T[0,1]
+        """
+        name = self._set_name(name, "matrix", axis_(axis, "code"))
+        fun = _ArrayAccess(axis_(axis, "index"))
+        needs = {Need.MATRIX}
+        super().__init__(fun, refpts, needs=needs, name=name, **kwargs)
+
+
+class _GlobalOpticsObservable(Observable):
```

Comment:
Because it's not supposed to used directly. If we want it public, on has to find new names for both `_GlobalOpticsObservable` and `GlobalOpticsObservable`.